openid-data-classes/src/commonMain/kotlin/at/asitplus/openid/RequestParametersFromSerializer.kt

```diff
@@ -0,0 +1,117 @@
+package at.asitplus.openid
+
+import at.asitplus.signum.indispensable.josef.JwsSigned
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.descriptors.element
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.JsonDecoder
+import kotlinx.serialization.json.JsonEncoder
+import kotlinx.serialization.json.buildJsonObject
+import kotlinx.serialization.json.decodeFromJsonElement
+import kotlinx.serialization.json.encodeToJsonElement
+import kotlinx.serialization.json.jsonObject
+
+/**
+ * In order to de-/serialize generic types we need a kind of factory approach
+ * Because we deal with a sealed class we can use an intermediary jsonSerializer,
+ * find the correct object and the specific type of the generic type and
+ * then finalize the serialization
+ *
+ * In order to de-/serialize JwsSigned which itself is again a generic class
+ * we use the fact that we can find the class of parameters before we need to know the
+ * generic class of JwsSigned. To serialize we use [JwsSignedSerializer].
+ */
+class RequestParametersFromSerializer<T : RequestParameters>(
+    private val parameterSerializer: KSerializer<T>,
+) : KSerializer<RequestParametersFrom<T>> {
+    override val descriptor: SerialDescriptor = buildClassSerialDescriptor("RequestParametersFromClass") {
+        element("Json", buildClassSerialDescriptor("Json") {
+            element<String>("jsonString")
+            element("parameters", parameterSerializer.descriptor)
```

Comment:
Better extract that string values to constants and also use them as `@SerialName` for the actual properties

Reply:
The classes actually dont need the serial name anymore. I made constants for the serializer itlself but removed them from `RequestParametersFrom`

Reply:
Since we use a JsonSerializer as in-between step we lose the class discriminator